docs/fsharp-cheatsheet.md

```diff
@@ -1,523 +1,630 @@
+// Create
+let array1 = [| "a"; "b"; "c" |]
+let array2 =
+    [| 1
+        2 |]
+let array3 = [| 1..2..9 |]  // start..increment..last; [| 1; 3; 5; 7; 9 |]
+
+// Indexed access
+let first = array1[0]  // "a"
 
-    // Slicing is inclusive; [| "a"; "b" |]
-    let firstTwo = array1[0..1]
+// Slicing is inclusive; [| "a"; "b" |]
+let firstTwo = array1[0..1]
 
-    // Assignment using `<-`
-    array1[1] <- "d"  // [| "a"; "d"; "c" |]
+// Assignment using `<-`
+array1[1] <- "d"  // [| "a"; "d"; "c" |]
 
-    // Pattern matching
-    match myArray with
-    | [||] -> ...        // match an empty array
-    | [| 3 |] -> ...     // match array with single 3 item
-    | [| _; 4 |] -> ...  // match array with 2 items, second item = 4
+// Pattern matching
+match myArray with
+| [||] -> ...        // match an empty array
+| [| 3 |] -> ...     // match array with single 3 item
+| [| _; 4 |] -> ...  // match array with 2 items, second item = 4
+```
 
 See the [Array Module](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-collections-arraymodule.html) for built-in functions.
 
 <div id="collections-sequences"></div>
 
-### Sequences
+## Sequences
 
 A *sequence* is a logical series of elements of the same type. `seq<'t>` is an alias for [`System.Collections.Generic.IEnumerable<'t>`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.ienumerable-1).
 
-    // Create
-    let seq1 = { 1; 2 }
-    let seq2 = seq {
-          1
-          2 }
-    let seq3 = seq { 1..2..9 }  // start..increment..last; 1,3,5,7,9
+```fsharp
+// Create
+let seq1 = { 1; 2 }
+let seq2 = seq {
+        1
+        2 }
+let seq3 = seq { 1..2..9 }  // start..increment..last; 1,3,5,7,9
+```
 
 See the [Seq Module](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-collections-seqmodule.html) for built-in functions.
 
-### Collection comprehension
+## Collection comprehension
 
 - Computed expressions with `->`. Results in _1, 3, 5, 7, 9_
-
-       let listComp  = [ for i in 0..4 -> 2 * i + 1 ]
-       let arrayComp = [| for i in 0..4 -> 2 * i + 1 |]
-       let seqComp   = seq { for i in 0..4 -> 2 * i + 1 }
+    ```fsharp
+    let listComp  = [ for i in 0..4 -> 2 * i + 1 ]
+    let arrayComp = [| for i in 0..4 -> 2 * i + 1 |]
+    let seqComp   = seq { for i in 0..4 -> 2 * i + 1 }
+    ```
 
 - Using computed expressions with `yield` and `yield!`. (`yield` is optional in a `do`, but is being used explicitly here):
-
-        let comprehendedList = [  // [ 1;3;5;7;9 ]
-            for i in 0..4 do
-                yield 2 * i + 1
-            ]
-        let comprehendedArray = [|  // [| 1;3;5;7;9;1;3;5;7;9 |]
-            for i in 0..4 do
-                yield 2 * i + 1
+    ```fsharp
+    let comprehendedList = [  // [ 1;3;5;7;9 ]
+        for i in 0..4 do
+            yield 2 * i + 1
+        ]
+    let comprehendedArray = [|  // [| 1;3;5;7;9;1;3;5;7;9 |]
+        for i in 0..4 do
+            yield 2 * i + 1
+        yield! listWithYield
```

Comment:
is this supposed to reference comprehendedList?

Reply:
Looks like it to me